draft-ietf-quic-tls.md

```diff
@@ -1276,10 +1276,30 @@ If 0-RTT keys are available (see {{enable-0rtt}}), the lack of replay protection
 means that restrictions on their use are necessary to avoid replay attacks on
 the protocol.
 
-A client MUST only use 0-RTT keys to protect data that is idempotent.  A client
-MAY wish to apply additional restrictions on what data it sends prior to the
-completion of the TLS handshake.  A client otherwise treats 0-RTT keys as
-equivalent to 1-RTT keys, except that it MUST NOT send ACKs with 0-RTT keys.
+Of the frames defined in {{QUIC-TRANSPORT}}, only the STREAM frame is
+potentially unsafe for use with 0-RTT as it carries application data.
```

Comment:
Let me nitpick, as you seem to be wondering about frames that might have issues.

I think that this sentence is slightly incorrect. There are other frames that also carry application data. Specifically, CONNECTION_CLOSE (type 0x1d) and RESET_STREAM carry information defined by the application, and we do not require the integers carried by those frames to be replay-safe. See #3440.

I think we should either discuss about "application data" in general, or name the three frames (STREAM, RESET_STREAM, CONNECTION_CLOSE (type 0x1d)) that carry information defined by the application.

Reply:
Excellent point.  Will amend.